### ISSUE TITLE
Allow extra fields via props

### ISSUE THREAD

**Issue by RamonGebben** [revolunet/react-mailchimp-subscribe#4]
It would be great if we would be able to specify the fields we want to submit, not only email.
Think fields like name, company etc.

**Comment by michalica** [comment]
is there possibility of adding extra field?

**Comment by revolunet** [comment]
Looks like it is possible on the mailchimp side so do you have an API in mind ?

As is, the component only provide the email. 

One could provide an array of fields as props ?

```js
const fields = [{
  name:"email",
  type:"text",
  placeholder:"Your email"
},{
  name:"company",
  type:"text",
  placeholder:"Company"
}]
``` 

this wont handle validation and more complex scenarios though

**Comment by RamonGebben** [comment]
@revolunet one could also provide a validation function which should return a Boolean on each field and maybe an error string for when validation fails. 

What kind of complex scenarios are you expecting to cause problems?

**Comment by revolunet** [comment]
say require phone 1 OR phone2 :)
or graphic customisation

maybe we should only do somthing like this :

```jsx
<MailchimpSubscribe url={mailchimpFormUrl} render={ ({ subscribe }) => (
    <MyCustomForm onValidated={ formData => subscribe(formData) }/>
) } />
```

So all the form logic and presentation is userland and we only provide the mailchimp POST

**Comment by revolunet** [comment]
Published a 2.0.0 with a render prop implementation, tell me what you think :)
